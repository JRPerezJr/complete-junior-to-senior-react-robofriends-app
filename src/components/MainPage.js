import React from 'react';

import Header from './Header';
import CardList from './CarList';
import Searchbox from './Searchbox';
import Scroll from './Scroll';
import ErrorBoundary from './ErrorBoundary';

import './MainPage.css';

class MainPage extends React.Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };
  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ? (
      <h1 className="tc">Loading...</h1>
    ) : (
      <div className="tc">
        <Header />
        <Searchbox searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundary>
            <CardList robots={this.filterRobots()} />
          </ErrorBoundary>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
