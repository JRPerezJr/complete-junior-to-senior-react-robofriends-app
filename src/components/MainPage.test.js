import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

describe('<MainPage/> component tests', () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false,
    };
    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it('sanity test', () => {
    expect(true).toBe(true);
  });

  it('renders MainPage component without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('filters robots correctly', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);
  });

  it('filters robots correctly 2', () => {
    const mockProps2 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'James',
          email: 'jbond@jbond.com',
        },
      ],
      searchField: 'james',
      isPending: false,
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);
    expect(wrapper2.instance().filterRobots()).toEqual([
      {
        id: 3,
        name: 'James',
        email: 'jbond@jbond.com',
      },
    ]);
  });

  it('filters robots correctly 3', () => {
    const mockProps3 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'James',
          email: 'jbond@jbond.com',
        },
      ],
      searchField: 'k',
      isPending: false,
    };

    const filteredRobots = [];
    const wrapper3 = shallow(<MainPage {...mockProps3} />);

    expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
  });

  it('should render <h1>Loading..</h1> if isPending true', () => {
    const mockProps4 = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'James',
          email: 'jbond@jbond.com',
        },
      ],
      searchField: 'k',
      isPending: true,
    };

    const wrapper4 = shallow(<MainPage {...mockProps4} />);

    expect(wrapper4.instance().render()).toMatchSnapshot();
  });
});
