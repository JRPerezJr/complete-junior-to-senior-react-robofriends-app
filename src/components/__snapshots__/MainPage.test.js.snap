// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage/> component tests renders MainPage component without crashing 1`] = `ShallowWrapper {}`;
